core/src/main/scala/com/microsoft/ml/spark/codegen/DotnetWrappable.scala

```diff
@@ -0,0 +1,522 @@
+
+  protected lazy val dotnetInternalWrapper = false
+
+  protected lazy val dotnetClassName: String = {
+    if (dotnetInternalWrapper) {
+      "_" + classNameHelper
+    } else {
+      "" + classNameHelper
+    }
+  }
+
+  protected def unCapitalize(name: String): String = {
+    Character.toLowerCase(name.charAt(0)) + name.substring(1)
+  }
+
+  protected lazy val dotnetClassNameString: String = s"s_${unCapitalize(dotnetClassName)}ClassName"
+
+  protected lazy val dotnetClassWrapperName: String = "WrapAs" + dotnetClassName
+
+  protected lazy val dotnetObjectBaseClass: String = {
+    thisStage match {
+      case _: Estimator[_] => s"ScalaEstimator<${companionModelClassName.split(".".toCharArray).last}>"
+      case _: Model[_] => s"ScalaModel<$dotnetClassName>"
+      case _: Transformer => s"ScalaTransformer"
+      case _: Evaluator => s"ScalaEvaluator"
+    }
+  }
+
+  protected def dotnetMLReadWriteMethods: String = {
+    s"""|/// <summary>
+        |/// Loads the <see cref=\"$dotnetClassName\"/> that was previously saved using Save(string).
+        |/// </summary>
+        |/// <param name=\"path\">The path the previous <see cref=\"$dotnetClassName\"/> was saved to</param>
+        |/// <returns>New <see cref=\"$dotnetClassName\"/> object, loaded from path.</returns>
+        |public static $dotnetClassName Load(string path) => $dotnetClassWrapperName(
+        |    SparkEnvironment.JvmBridge.CallStaticJavaMethod($dotnetClassNameString, "load", path));
+        |
+        |/// <summary>
+        |/// Saves the object so that it can be loaded later using Load. Note that these objects
+        |/// can be shared with Scala by Loading or Saving in Scala.
+        |/// </summary>
+        |/// <param name="path">The path to save the object to</param>
+        |public void Save(string path) => Reference.Invoke("save", path);
+        |
+        |/// <returns>a <see cref=\"ScalaMLWriter\"/> instance for this ML instance.</returns>
+        |public ScalaMLWriter Write() =>
+        |    new ScalaMLWriter((JvmObjectReference)Reference.Invoke("write"));
+        |
+        |/// <returns>an <see cref=\"ScalaMLReader\"/> instance for this ML instance.</returns>
+        |public ScalaMLReader<$dotnetClassName> Read() =>
+        |    new ScalaMLReader<$dotnetClassName>((JvmObjectReference)Reference.Invoke("read"));
+        |""".stripMargin
+  }
+
+  protected def dotnetWrapAsTypeMethod: String = {
+    s"""|private static $dotnetClassName $dotnetClassWrapperName(object obj) =>
+        |    new $dotnetClassName((JvmObjectReference)obj);
+        |""".stripMargin
+  }
+
+  def dotnetAdditionalMethods: String = {
+    ""
+  }
+
+  //noinspection ScalaStyle
+  protected def dotnetParamSetter(p: Param[_]): String = {
+    val capName = p.name match {
+      case "xgboostDartMode" => "XGBoostDartMode"
+      case "parallelism" => dotnetClassName match {
+        case "VowpalWabbitContextualBandit" => "ParallelismForParamListFit"
```

Comment:
Could you explain why these things need to be here? Perhaps a comment would help